### DISCUSSION TITLE
`predict_eval_analyse.py` error: AssertionError: Results do not correspond to current coco set

### DISCUSSION THREAD

**Discussion by little-spoon** [obss/sahi#717] [Q&A] [answered]
When i use `predict_eval_analyse.py` :
```shell
Performing inference on images: 100%|█████████| 262/262 [31:06<00:00,  7.12s/it]
Prediction results are successfully exported to runs/predict_eval_analyse/XVIEW_SAHI_FI_PO3
loading annotations into memory...
Done (t=0.04s)
creating index...
index created!

Evaluating bbox...
Loading and preparing results...
Traceback (most recent call last):
  File "predict_evaluate_analyse.py", line 115, in <module>
    return_dict=False,
  File "/home/brl/anaconda3/envs/xsd_t/lib/python3.7/site-packages/sahi/scripts/coco_evaluation.py", line 372, in evaluate
    COCOeval=COCOeval,
  File "/home/brl/anaconda3/envs/xsd_t/lib/python3.7/site-packages/sahi/scripts/coco_evaluation.py", line 128, in evaluate_core
    cocoDt = cocoGt.loadRes(results)
  File "/home/brl/anaconda3/envs/xsd_t/lib/python3.7/site-packages/pycocotools/coco.py", line 328, in loadRes
    'Results do not correspond to current coco set'
AssertionError: Results do not correspond to current coco set
```
How can i solve this problems? Thank you!

**Comment by fcakyon** [comment]
This error means your dataset.json and result.json have different categories. You have to fix this issue before evaluating it.

Are your category ids start from 0 in the dataset.json?

You can use the coco utils present in sahi to update your category ids: https://github.com/obss/sahi/discussions/336#discussioncomment-1918853

**Comment by little-spoon** [comment]
I find my category ids start from 1!
So I need to fix this, right?






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "obss/sahi"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月5日(星期六) 下午4:38
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [obss/sahi] `predict_eval_analyse.py` error: AssertionError: Results do not correspond to current coco set (Discussion #717)





 
This error means your dataset.json and result.json have different categories. You have to fix this issue before evaluating it.
 
Are your category ids start from 0 in the dataset.json?

**Reply by fcakyon** [reply]
Yes, you can use the coco utils present in sahi to update your category ids: https://github.com/obss/sahi/discussions/336#discussioncomment-1918853

**Comment by fcakyon** [comment, marked as answer]
Your category ids should start from 0 in the dataset.json.

You can use the coco utils present in sahi to update your category ids: https://github.com/obss/sahi/discussions/336#discussioncomment-1918853

**Comment by little-spoon** [comment]
I see!
Thank you!


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "obss/sahi"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月5日(星期六) 下午4:55
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [obss/sahi] `predict_eval_analyse.py` error: AssertionError: Results do not correspond to current coco set (Discussion #717)





 
Your category ids start from 0 in the dataset.json.
 
You can use the coco utils present in sahi to update your category ids: #336 (reply in thread)

**Reply by fcakyon** [reply]
you're welcome, have a nice day!
